moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -463,28 +462,53 @@ KinematicState Servo::getNextJointState(const ServoInput& command)
   return target_state;
 }
 
-const TwistCommand Servo::toPlanningFrame(const TwistCommand& command)
+Eigen::Isometry3d Servo::getPlanningToCommandFrameTransform(const std::string& command_frame) const
 {
-  Eigen::Vector<double, 6> transformed_twist = command.velocities;
+  const moveit::core::RobotStatePtr robot_state = planning_scene_monitor_->getStateMonitor()->getCurrentState();
+  if (robot_state->knowsFrameTransform(command_frame))
+  {
+    return robot_state->getGlobalLinkTransform(servo_params_.planning_frame).inverse() *
```

Comment:
Do we need this?
```
robot_state->getGlobalLinkTransform(servo_params_.planning_frame).inverse() *
```
Doesn't `getGlobalLinkTransform(command.frame_id)` already give transform w.r.t to `planning_frame` ?

```
Get the link transform w.r.t. the root link (model frame) of the RobotModel.
This is typically the root link of the URDF unless a virtual joint is present.
```

Reply:
Is the URDF root link necessarily the base link of the planning group? I would think not.

Also, there's the note about the virtual joint you pasted in as a (distant) second reason to want to keep this?

Reply:
I see, in that case , better to keep it :+1: